sandbox/find_conflicting_files.py

```diff
@@ -0,0 +1,34 @@
+import os
+import json
+import filecmp
+import re
+
+
+def find_revision(path):
+    with open(path, 'r') as f:
+        p = re.compile(r'\s*revision\s+"\d\d\d\d-\d\d-\d\d"')
+        for line in f:
+            m = p.match(line)
+            if m:
+                return m.group()[-11:-1]
+    return '1970-01-01'
+
+
+conflicting = []
+fnames = {}
+top = '/backend/tests/resources/yangmodels/yang'
+for dirname, _, files in os.walk(top):
+    dirname = os.path.join(top, dirname)
+    for f in files:
+        if f.endswith('.yang'):
+            revision = find_revision(os.path.join(dirname, f))
```

Comment:
This doesn t make it for sure that the revision is the last one. If you check the RFC it is said that each yang file can have several revisions as the yang file is updated.. it should be in order from newer to older but there are many cases that this is not true. 

In your find revision case you find the first one and return it. the parser that I proposed takes care of such cases that is why it s better to use that.

Reply:
The parser was consuming >7GB of RAM and freezing up my machine when I tried it. I'll try and get the latest revision with re if that's ok.

Reply:
on which module was it?

make sure it s not just building up the schema context but you keep creating a new one.. otherwise with each module you are taking more and more memory..

Reply:
oh I see, how would I prevent this?